hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientInvocationFuture.java

```diff
@@ -39,11 +43,10 @@
     private final CallIdSequence callIdSequence;
 
     public ClientInvocationFuture(ClientInvocation invocation,
-                                  Executor internalExecutor,
                                   ClientMessage request,
                                   ILogger logger,
                                   CallIdSequence callIdSequence) {
-        super(internalExecutor, logger);
+        super(logger);
```

Comment:
Here we changed the executor that `callbacks` are running on. This will change the behavior. 
For example, when an instance is down,  If we run a callback on a future we will not get RejectedExecutionException.

Reply:
Yes, the goal is to unify behaviour across member & client and align with JDK's `CompletableFuture` that executes callbacks registered with default `*Async` methods on the `ForkJoinPool#commonPool`. So default executor for member-side `CompletableFuture`s is no longer `asyncExecutor` and client-side `CompletableFuture`s respectively no longer use the `userExecutor` by default.